### PR TITLE
vk: add workaround for Intel HD/UHD memory leak (#117)

### PR DESCRIPTION
**Pull request by skejeton**
This will detect an Intel graphics card and apply the fix for leaky descriptor pools.

### CODE REVIEW

File: blade-graphics/src/vulkan/init.rs [hunk 1]
```diff
@@ -143,6 +150,9 @@ unsafe fn inspect_adapter(
     let name = ffi::CStr::from_ptr(properties.device_name.as_ptr());
     log::info!("Adapter {:?}", name);
 
+    let vendor_id = properties2_khr.properties.vendor_id;
+    let intel_fix_descriptor_pool_leak = vendor_id == db::intel::VENDOR;
```
Comment (kvark) [inline blade-graphics/src/vulkan/init.rs:154]: can we limit to Windows only?

File: blade-graphics/src/vulkan/init.rs [hunk 2]
```diff
@@ -143,6 +150,9 @@ unsafe fn inspect_adapter(
     let name = ffi::CStr::from_ptr(properties.device_name.as_ptr());
     log::info!("Adapter {:?}", name);
 
+    let vendor_id = properties2_khr.properties.vendor_id;
```
Comment (kvark) [inline blade-graphics/src/vulkan/init.rs:153]: if we are doing this, let's also use `vendor_id` when computing `intel_unable_to_present` consistently

File: blade-graphics/src/vulkan/mod.rs [hunk 3]
```diff
@@ -23,6 +23,7 @@ struct RayTracingDevice {
 struct Workarounds {
     extra_sync_src_access: vk::AccessFlags,
     extra_sync_dst_access: vk::AccessFlags,
+    intel_fix_descriptor_pool_leak: bool,
```
Comment (kvark) [inline blade-graphics/src/vulkan/mod.rs:26]: Perhaps we can instead store it as
```
extra_descriptor_pool_create_flags: vk::DescriptorPoolCreateFlags
```
Or at least call it something neutral, like `force_free_descriptor_set_flag`

File: blade-graphics/src/vulkan/descriptor.rs [hunk 4]
```diff
@@ -45,10 +45,19 @@ impl super::Device {
...
             ..Default::default()
         };
-        let descriptor_pool_info = vk::DescriptorPoolCreateInfo::default()
-            .max_sets(max_sets)
-            .pool_sizes(&descriptor_sizes)
-            .push_next(&mut inline_uniform_block_info);
+        let descriptor_pool_info = if self.workarounds.intel_fix_descriptor_pool_leak {
+            vk::DescriptorPoolCreateInfo::default()
+                .max_sets(max_sets)
+                .flags(vk::DescriptorPoolCreateFlags::FREE_DESCRIPTOR_SET)
```
Comment (kvark) [inline blade-graphics/src/vulkan/descriptor.rs:51]: this if/else should be moved inside the `flags`
Either
```rust
.flags(self.workarounds.extra_descriptor_pool_create_flags)
```
Or
```rust
.flags(if ... {} else {})
```

**Review by kvark — COMMENTED** [review]
Thank you for the PR!
A few small changes are needed, but overall the approach is good

**Comment by skejeton** [comment]
fixed

**Review by kvark — APPROVED** [review]
great, thank you!

**Pull request merged**